### DISCUSSION TITLE
ScriptReference/AssetDatabase

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#278] [Notes]
# ScriptReference/AssetDatabase

https://docs.unity3d.com/ScriptReference/AssetDatabase.html

**Comment by docscomments** [comment]
**Comment by `@ByronMayne`:**  

Working with folders in Unity is a super huge pain. If you are creating folders with `System.IO` this will be imported into Unity later on. So if you create a folder and also try to add stuff to it in code you are going to have issues. This is why Unity has their own functions, to create and import them in one step. I have written an free open source library that you can do that makes these task much easier.  
[Unity IO](https://github.com/ByronMayne/UnityIO)

**Comment by rhys-vdw** [comment]
There is also an undocumented method `GetAllAssetPaths` which, as its name suggests, returns the path of all assets. Not sure why it's not documented, it is not deprecated.
